src/main/java/org/onedatashare/server/module/vfs/VfsResource.java

```diff
@@ -214,7 +214,8 @@ public Long buildDirectoryTree(List<Stat> sub, FileObject[] fileObjects, String
                         if (state + sliceSizeInt < sizeInt) {
                             byte[] b = new byte[sliceSizeInt];
                             try {
-                                finalInputStream.read(b, 0, sliceSizeInt);
+                                for(int offset = 0; offset < sliceSizeInt; offset+=1)
+                                    finalInputStream.read(b, offset, 1);
```

Comment:
Reading chunks doesn't work, the PDF files get corrupted for some reason. This is a temporary fix for the time. Haven't been able to figure out what's causing the corruption exactly.

Reply:
Hmm..

2 possible suggestions -
1. If the issue is only in the last chunk that gets read, then only having this logic in the else block makes sense.
2. Rather than having all files go through this and slowing the process, have a conditional check if the file is a pdf and route it to this logic.

I would advise to discuss this out with the team since I am not completely aware of the issue.
Also, you can work on this on a new branch/PR. For now we can let this pass in this PR.

Reply:
The issue happens whenever there is a special EOF character in the middle of the file. Does not limit to the last chunk and does not limit to PDF. But pdf seems to have this character often. I think the problem occurs because of VFS commons implementation is sensitive to such EOF character.

Reply:
Hmm.. Sounds like a valid issue.
I was not a fan of Apache VFS since the beginning. :D

Reply:
Strangely, it works perfectly fine when I use the commons VFS to download the file to the disk. I tried using another library for Http Downloads for the transfers and I encounter the same error I faced with VFS.